pkgs/tools/graphics/mangohud/default.nix

```diff
@@ -100,13 +80,15 @@ in stdenv.mkDerivation rec {
   postUnpack = ''(
     cd "$sourceRoot/subprojects"
     cp -R --no-preserve=mode,ownership ${imgui.src} imgui-${imgui.version}
-    cp -R --no-preserve=mode,ownership ${spdlog.src} spdlog-${spdlog.version}
     cp -R --no-preserve=mode,ownership ${vulkan-headers.src} Vulkan-Headers-${vulkan-headers.version}
   )'';
 
-  env.NIX_CFLAGS_COMPILE = "-I${vulkan-headers.src}/include";
-
   patches = [
+    # Add @libraryPath@ template variable to fix loading the preload
+    # library and @dataPath@ to support overlaying Vulkan apps without
+    # requiring MangoHud to be installed
+    ./preload-nix-workaround.patch
+
```

Comment:
I based this new approach around the same thing we're doing for gamemode: https://github.com/NixOS/nixpkgs/blob/2ba48cc8dee2249526efeda1d2c629a2ec625229/pkgs/tools/games/gamemode/default.nix#L33-L34